extensions/ext/GL_EXT_null_initializer.txt

```diff
@@ -0,0 +1,150 @@
+Name
+
+  EXT_null_initializer
+
+Name Strings
+
+  GL_EXT_null_initializer
+
+Contact
+
+  Alan Baker, Google (alanbaker 'at' google.com)
+
+Contributors
+
+  Alan Baker, Google
+  Jeff Bolz, NVIDIA
+  Jason Ekstrand, Intel
+  John Kessenich, Google
+  Daniel Koch, NVIDIA
+
+Status
+
+  Draft
+
+Version
+
+  Last Modified Date:      June 11, 2020
+  Revision:                3
+
+Number
+
+  TBD
+
+Depenendencies
+
+  This extension can be applied to OpenGL GLSL versions 1.40
+  (#version 140) and higher.
+
+  This extension can be applied to OpenGL ES ESSL versions 3.10
+  (#version 310) and higher.
+
+  This extension is written against the OpenGL Shading Language
+  Specification, version 4.60 (revision 7), dated July 10, 2019
+
+  This extension requires GL_KHR_vulkan_glsl
+
+Overview
+
+  This extension adds a null initializer to GLSL and allow shared variables in
+  compute shaders to be null initiailzed. Null initializers are specified using
+  a new syntax of empty braces.
+
+New Procedures and Functions
+
+  None.
+
+New Tokens
+
+  None.
+
+Modifications to GL_KHR_vulkan_glsl
+
+  Add to the "Mapping to SPIR-V" section
+
+  Mapping shared variable initializers:
+
+    Null initializers should be generated as OpConstantNull of the variable's type.
+
+Modifications to the OpenGL Shading Language Specification, Version 4.60
+
+  Including the following line in a shader can be used to control the language
+  features described in this extension:
+
+    #extension GL_EXT_null_initialize_shared_memory : <behavior>
+
+  where <behavior> is as specified in section 3.3.
+
+  New preprocessor #defines are added to the OpenGL Shading Language:
+
+    #define GL_EXT_null_initialize_shared_memory    1
+
+  Add a new section 4.1.11.1 Null Initializers:
+
+    A null initializer is defined for the following types:
+
+    - bool: false
+    - Scalar integers: 0
+    - Scalar floating point: +0.0 (all bits 0)
+    - Composites: each member is recursively defined to be null according to their
+      constituent types
+
+   Null initialization can be achieved by using an empty brace expression. Null
+   initializers may only be used to initialize sized-types.
```

Comment:
I'm not sure what "sized types" here means. (Also, I'm not sure it should be hyphenated). Are you disallowing opaque types (that don't have a fixed size in memory) or unsized arrays? Or both?

The spec also tries to say when errors will occur if the rules are not followed, ie. whether it's a compile- or link-time error.

```suggestion
   It is a compile-time error to null initialize any other type, or to null initialize an
   unsized array type.
   
   Null initialization can be achieved by using an empty brace expression.
```